crates/weaver_semconv/src/registry.rs

```diff
@@ -64,20 +65,37 @@ impl SemConvRegistry {
     /// # Errors
     ///
     /// If the registry path pattern is invalid.
-    pub fn try_from_path_pattern(registry_id: &str, path_pattern: &str) -> Result<Self, Error> {
-        let mut registry = SemConvRegistry::new(registry_id);
-        for sc_entry in glob::glob(path_pattern).map_err(|e| Error::InvalidRegistryPathPattern {
-            path_pattern: path_pattern.to_owned(),
-            error: e.to_string(),
-        })? {
-            let path_buf = sc_entry.map_err(|e| Error::InvalidRegistryPathPattern {
-                path_pattern: path_pattern.to_owned(),
-                error: e.to_string(),
-            })?;
-            let semconv_spec = SemConvSpecWithProvenance::from_file(path_buf.as_path())?;
-            registry.add_semconv_spec(semconv_spec);
+    pub fn try_from_path_pattern(registry_id: &str, path_pattern: &str) -> WResult<Self, Error> {
+        fn create_registry_or_fatal(
```

Comment:
This pattern simplifies the management of fatal errors using a simple `?` operator. This internal method populates a list of non-fatal errors and returns a `Result`. This `Result` is then transformed into a `WResult` depending on the presence or absence of non-fatal errors.

Reply:
This gets around limitations in the current expeirmental custom type `?` support for sure.  I'd still like to go that direction with WResult when it stabilizes, but for now this seems to be our best bet.

I may take some time to write up a guide on using WResult that includes this, but sadly I've been highly limited on time recently (possibly when entities OTEP has progressed past prototyping, I can come back and revisit this)